spec/lib/user_spec.rb

```diff
@@ -0,0 +1,97 @@
+RSpec.describe User do
+  after do
+    DB[:userdetails].truncate
+  end
+
+  describe '#generate' do
+    context 'new user' do
+      before do
+        stub_const('User::WORD_LIST', word_list)
+      end
+
+      let(:random_username) do
+        srand(2)
+        ('a'..'z').to_a.sample(6).join
+      end
+
+      let(:word_list) { %w[These Are Words] }
+      let(:email) { 'foo@bar.gov.uk' }
+
+      let!(:user) do
+        srand(2)
+        User.new.generate(email: email)
+      end
+
+      let(:username_password_from_db) { User.select(:username, :password).first.values }
+      let(:user_from_db) { User.first }
+      let(:split_password) { user[:password].split(/(?=[A-Z])/) }
+
+      it 'creates a user and returns it' do
+        expect(user[:username]).to eq(random_username)
+        expect(user[:password]).not_to be_empty
+      end
+
+      it 'creates a random password from the word list' do
+        expect(split_password.sort).to eq(word_list.sort)
+      end
+
+      it 'stores the user in the database' do
+        expect(username_password_from_db).to eq(user)
+      end
+
+      it 'stores the email as both the email and sponsor for the user' do
+        expect(user_from_db.email).to eq(email)
+        expect(user_from_db.sponsor).to eq(email)
+      end
+    end
+  end
+
+  context 'avoiding duplicate usernames' do
+    before do
+      srand(2)
+      ('a'..'z').to_a.sample(6).join
```

Comment:
What's this line do?

Reply:
But the value isn't assigned anywhere which is what confuses me.

Reply:
Gotcha, removed it